openmc/mgxs_library.py

```diff
@@ -457,102 +340,262 @@ def use_chi(self, use_chi):
+        ----------
+        total: np.ndarray
+            Total Cross Section
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset. Defaults
+            to 294K
+
+        See also
+        --------
+        openmc.mgxs_library.set_total_mgxs()
+
+        """
         check_type('total', total, Iterable, expected_iter_type=Real)
-        # Convert to a numpy array so we can easily get the shape for
-        # checking
+        # Convert to a numpy array so we can easily get the shape for checking
         nptotal = np.asarray(total)
         check_value('total shape', nptotal.shape, [self.vector_shape])
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = self.temperatures.tolist().index(temperature)
+        self._total[i] = nptotal
 
-        self._total = nptotal
+    def set_absorption(self, absorption, temperature=294.):
+        """This method sets the cross section for this XSdata object at the
+        provided temperature.
+
+        Parameters
+        ----------
+        absorption: np.ndarray
+            Absorption Cross Section
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset. Defaults
+            to 294K
+
+        See also
+        --------
+        openmc.mgxs_library.set_absorption_mgxs()
 
-    @absorption.setter
-    def absorption(self, absorption):
+        """
         check_type('absorption', absorption, Iterable, expected_iter_type=Real)
-        # Convert to a numpy array so we can easily get the shape for
-        # checking
+        # Convert to a numpy array so we can easily get the shape for checking
         npabsorption = np.asarray(absorption)
         check_value('absorption shape', npabsorption.shape,
                     [self.vector_shape])
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = self.temperatures.tolist().index(temperature)
+        self._absorption[i] = npabsorption
+
+    def set_fission(self, fission, temperature=294.):
+        """This method sets the cross section for this XSdata object at the
+        provided temperature.
+
+        Parameters
+        ----------
+        fission: np.ndarray
+            Fission Cross Section
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset. Defaults
+            to 294K
 
-        self._absorption = npabsorption
+        See also
+        --------
+        openmc.mgxs_library.set_fission_mgxs()
 
-    @fission.setter
-    def fission(self, fission):
+        """
         check_type('fission', fission, Iterable, expected_iter_type=Real)
-        # Convert to a numpy array so we can easily get the shape for
-        # checking
+        # Convert to a numpy array so we can easily get the shape for checking
         npfission = np.asarray(fission)
```

Comment:
OH i remember why i did this.  The user of this library passed absorption, which could be a list.  Its not my call to change what they passed me, so I introduced npfission (etc) to simply be a local version I can change.